crud/src/main/java/com/redhat/lightblue/assoc/ep/ExecutionPlan.java

```diff
@@ -0,0 +1,474 @@
+                        search.setLimit(from,to);
+                        search.setSort(requestSort);
+                        sorted=limited=true;
+                    } else {
+                        // There are unassigned clauses. We can sort during search, but we have to filter and limit
+                        search.setSort(requestSort);
+                        sorted=true;
+                        Filter f=new Filter(block,new Source<>(search),getFilterQuery(unassigned));
+                        last=new Source<>(f);
+                        if(from!=null)
+                            last=new Source<>(new Skip(block,from.intValue(),last));
+                        if(to!=null)
+                            last=new Source<>(new Limit(block,to.intValue()-from.intValue()+1,last));
+                        limited=true;
+                    }
+                }
+                search.recordResultSetSize(true);
+                
+                Set<Path> fields=getIncludedFieldsOfEntityForSearch(block,qfi);
+                fields.addAll(getIncludedFieldsOfEntityForProjection(block,rootMd,requestProjection));
+                search.setProjection(writeProjection(fields));
+                search.setQueries(node.getData().getConjuncts());
+                resultStep=new Assemble(block,last,destinationBlocks);
+                resultStep=new Project(block,new Source<>(resultStep),requestProjection);
+                block.setResultStep(resultStep);
+            } else {
+                // Processing one of the associated entity nodes
+                // Reuse the batching algorithm in join                
+                ExecutionBlock source=block.getSourceBlocks().get(0);
+                QueryPlanData edgeData=retrievalQueryPlan.getEdgeData(source.getQueryPlanNode(),
+                                                                      block.getQueryPlanNode());
+                if(edgeData!=null) {
+                    AssociationQuery aq=new AssociationQuery(rootMd,
+                                                             block.getMetadata(),
+                                                             block.getReference(),
+                                                             edgeData.getConjuncts());
+                    block.setAssociationQuery(source,aq);
+                }
+                Retrieve search=new Retrieve(block);
+                search.setQueries(node.getData().getConjuncts());
+                Set<Path> fields=getIncludedFieldsOfEntityForSearch(block,qfi);
+                fields.addAll(getIncludedFieldsOfEntityForProjection(block,rootMd,requestProjection));
+                search.setProjection(writeProjection(fields));
+                block.setResultStep(new Assemble(block,new Source<>(search),destinationBlocks));
+            }
+        }
+
+        for(ExecutionBlock block:qp2BlockMap.values()) {
+            block.initialize();
+        }
+    }
```

Comment:
Hm.  I'll take a better look through it after I get to the end and see if I have any ideas.